File: app/src/main/java/com/hhy/game/dodgebox/game/Player.java
package com.hhy.game.dodgebox.game;

import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Matrix;
import android.graphics.Rect;

class Player {
    private Animator animator;
    private Bitmap bitmap;
    private Values values;
    private int width;
    private int height;
    private VectorXY coordinates;
    // true is left, false is right
    private boolean left = false;

    Player(Bitmap image, Values values, boolean isDead) {
        this.values = values;
        bitmap = image;
        // two diffirent images for player for now
        // die images & run images
        if (isDead) {
            width = bitmap.getWidth() / 3;
            height = bitmap.getHeight() / 2;
        } else {
            width = bitmap.getWidth() / 4;
            height = bitmap.getHeight() / 3;
        }

        // starting coordinates
        coordinates = new VectorXY(values.getWidth() / 2, values.getGroundY() - height);
    }

    // run to left
    void moveLeft() {
        int X = coordinates.getX() - 10;
        if (X < 0) X = 0;
        coordinates.set(X, coordinates.getY());
    }

    // run to right
    void moveRight() {
        int X = coordinates.getX() + 10;
        if (X > values.getWidth() - width) X = values.getWidth() - width;
        coordinates.set(X, coordinates.getY());
    }

    void update() {

    }

    // draw the character
    void draw(Canvas canvas) {
        canvas.drawBitmap(bitmap, coordinates.getX(), coordinates.getY(), null);
    }

    // set coordinates
    void setCoordinates(VectorXY coordinates) {
        this.coordinates = coordinates;
    }

    // get coordinates
    VectorXY getCoordinates() {
        return coordinates;
    }

    // character rect angle
    Rect getBoundingRectangle() {
        return new Rect(coordinates.getX() + width / 5, coordinates.getY() + height / 5,
                coordinates.getX() + width / 5 * 4, coordinates.getY() + height / 5 * 4);
    }

    // turn left
    void setLeft(boolean left) {
        this.left = left;
    }

    // for run to left
    private void rotateBitmap(Bitmap bitmap) {
        Matrix matrix = new Matrix();
        matrix.postScale(-1, 1, bitmap.getWidth() / 2f, bitmap.getHeight() / 2f);
        this.bitmap = Bitmap.createBitmap(bitmap, 0, 0, bitmap.getWidth(), bitmap.getHeight(), matrix, true);
    }

    // for animating character
    void setBitmap(Bitmap bitmap) {
        if (left) rotateBitmap(bitmap);
        else this.bitmap = bitmap;
    }

    // create animator
    void createAnimator(Bitmap bitmapWithFrames,
                               int numberOfFramesHorizontally,
                               int numberOfFramesVertically, int updateTimeMillis,
                               double timeForOneFrameMillis) {
        if (animator == null)
            animator = new Animator(this, bitmapWithFrames, numberOfFramesHorizontally, numberOfFramesVertically, updateTimeMillis, timeForOneFrameMillis);
    }

    // not using for now but who knows
    void destroyAnimator() {
        animator.stopAnimation();
        animator = null;
    }

    // start animation
    void startAnimation() {
        if ((animator != null) && (!animator.isRunning()))
            animator.startAnimation();
    }

    // stop animation
    void stopAnimation() {
        if ((animator != null) && (animator.isRunning()))
            animator.stopAnimation();
    }
}

File: app/src/main/java/com/hhy/game/dodgebox/game/Player.kt
package com.hhy.game.dodgebox.game

import android.graphics.Bitmap
import android.graphics.Canvas
import android.graphics.Matrix
import android.graphics.Rect

internal class Player(private var bitmap: Bitmap, private val values: Values, isDead: Boolean) {
    private lateinit var animator: Animator
    private var width = 0
    private var height = 0

    // set coordinates
    // get coordinates
    var coordinates: VectorXY

    // true is left, false is right
    private var left = false

    // run to left
    fun moveLeft() {
        var x = coordinates.x - 10
        if (x < 0) x = 0
        coordinates[x] = coordinates.y
    }

    // run to right
    fun moveRight() {
        var x = coordinates.x + 10
        if (x > values.width - width) x = values.width - width
        coordinates[x] = coordinates.y
    }

    fun update() {}

    // draw the character
    fun draw(canvas: Canvas) {
        canvas.drawBitmap(bitmap, coordinates.x.toFloat(), coordinates.y.toFloat(), null)
    }

    // character rect angle
    val boundingRectangle: Rect
        get() = Rect(coordinates.x + width / 5, coordinates.y + height / 5,
                coordinates.x + width / 5 * 4, coordinates.y + height / 5 * 4)

    // turn left
    fun setLeft(left: Boolean) {
        this.left = left
    }

    // for run to left
    private fun rotateBitmap(bitmap: Bitmap) {
        val matrix = Matrix()
        matrix.postScale(-1f, 1f, bitmap.width / 2f, bitmap.height / 2f)
        this.bitmap = Bitmap.createBitmap(bitmap, 0, 0, bitmap.width, bitmap.height, matrix, true)
    }

    // for animating character
    fun setBitmap(bitmap: Bitmap) {
        if (left) rotateBitmap(bitmap) else this.bitmap = bitmap
    }

    // create animator
    fun createAnimator(bitmapWithFrames: Bitmap,
                       numberOfFramesHorizontally: Int,
                       numberOfFramesVertically: Int, updateTimeMillis: Int,
                       timeForOneFrameMillis: Double) {
        animator = Animator(this, bitmapWithFrames, numberOfFramesHorizontally, numberOfFramesVertically, updateTimeMillis, timeForOneFrameMillis)
    }

    // not using for now but who knows
    fun destroyAnimator() {
        animator.stopAnimation()
    }

    // start animation
    fun startAnimation() {
        if (!animator.isRunning) animator.startAnimation()
    }

    // stop animation
    fun stopAnimation() {
        if (animator.isRunning) animator.stopAnimation()
    }

    init {
        // two diffirent images for player for now
        // die images & run images
        if (isDead) {
            width = bitmap.width / 3
            height = bitmap.height / 2
        } else {
            width = bitmap.width / 4
            height = bitmap.height / 3
        }

        // starting coordinates
        coordinates = VectorXY(values.width / 2, values.groundY - height)
    }
}